src/test/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepTest.java

```diff
@@ -239,6 +241,24 @@ public void evaluate() throws Throwable {
         });
     }
 
+    @Test
+    public void activityRemote() {
+        assumeFalse(Functions.isWindows()); // TODO create analogue using bat
+        story.then(r -> {
+            r.createSlave();
+            WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition("" +
+                     "node('!master') {\n" +
+                     "  timeout(time:5, unit:'SECONDS', activity: true) {\n" +
+                     "    sh 'set +x; echo NotHere; sleep 3; echo NotHereYet; sleep 3; echo JustHere; sleep 10; echo ShouldNot'\n" +
```

Comment:
Cannot use `echo` steps as in the other tests, because we need to demonstrate serialization of the `ConsoleLogFilter` impl, which only happens if a nested step actually remotes its `TaskListener`.